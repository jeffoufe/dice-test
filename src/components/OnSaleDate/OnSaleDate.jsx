// @flow

import React from 'react';
import styles from './OnSaleDate.module.scss';
import moment from 'moment';

type OnSaleDateProps = {
    date: string
}

const OnSaleDate = ({ date }: OnSaleDateProps) => (
    <div className={styles.container}>
        <div className={styles.date}>
            {`On sale ${moment(date).format('D MMM LT')}`}
        </div>
    </div>
)

export default OnSaleDate;