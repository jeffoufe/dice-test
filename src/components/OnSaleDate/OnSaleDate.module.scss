@import "../../constants/colors";

.on-sale {
    &-container {
        width: 168px;
        height: 23px;
        padding: 2px 5px 4px;
        font-weight: 600;
        position: absolute;
        right: 17px;
        bottom: 17px;
        font-size: 14px;
        line-height: 16px;
        letter-spacing: 0.05em;
        color: white;
        background-color: $black;
    }

    &-date {
        display: inline-block;
        vertical-align: -webkit-baseline-middle;
    }
}