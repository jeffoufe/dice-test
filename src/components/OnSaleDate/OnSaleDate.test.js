import OnSaleDate from './OnSaleDate';
import { shallow } from 'enzyme';
import React from 'react';

describe('OnSaleDate', () => {
    const props = {
        date: 'Mon Aug 19 2019 21:54:38'
    };

    const wrapper = shallow(
        <OnSaleDate {...props} />
    );

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
})