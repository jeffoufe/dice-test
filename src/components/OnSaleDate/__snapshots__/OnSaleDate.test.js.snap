// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnSaleDate renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="date"
  >
    On sale 19 Aug 9:54 PM
  </div>
</div>
`;
